tests/bluetooth/tester/src/btp_vcp.c

```diff
@@ -0,0 +1,587 @@
+/* btp_vcp.c - Bluetooth VCP Tester */
+
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/testing.h>
+#include <zephyr/bluetooth/audio/vcp.h>
+#include <zephyr/bluetooth/audio/aics.h>
+#include "zephyr/bluetooth/audio/vocs.h"
+
+#include <app_keys.h>
+
+#include <zephyr/logging/log.h>
+#define LOG_MODULE_NAME bttester_vcp
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include "btp/btp.h"
+
+#define CONTROLLER_INDEX 0
+#define BT_AICS_MAX_INPUT_DESCRIPTION_SIZE 16
+#define BT_AICS_MAX_OUTPUT_DESCRIPTION_SIZE 16
+
+struct bt_vcp_vol_rend_register_param vcp_register_param;
+struct bt_vcp_included included;
+
+static void set_register_params(uint8_t gain_mode);
+uint8_t tester_init_vcp(void);
+
+
+/* Volume Control Service */
+static void vcs_supported_commands(uint8_t *data, uint16_t len)
+{
+	struct net_buf_simple *buf = NET_BUF_SIMPLE(BTP_DATA_MAX_SIZE);
+
+	net_buf_simple_init(buf, 0);
+	net_buf_simple_add_u8(buf, VCS_READ_SUPPORTED_COMMANDS);
+	net_buf_simple_add_u8(buf, VCS_SET_VOL);
+	net_buf_simple_add_u8(buf, VCS_VOL_UP);
+	net_buf_simple_add_u8(buf, VCS_VOL_DOWN);
+	net_buf_simple_add_u8(buf, VCS_MUTE);
+	net_buf_simple_add_u8(buf, VCS_UNMUTE);
+
+	tester_send(BTP_SERVICE_ID_VCS, VCS_READ_SUPPORTED_COMMANDS,
+		    CONTROLLER_INDEX, buf->data, buf->len);
+}
+
+static void set_volume(uint8_t *data)
+{
+	const struct vcs_set_vol_cmd *cmd = (void *)data;
+	uint8_t volume;
+	int err;
+
+	volume = cmd->volume;
+
+	LOG_DBG("volume 0x%02x", volume);
+
+	err = bt_vcp_vol_rend_set_vol(volume);
+	if (err) {
+		tester_rsp(BTP_SERVICE_ID_VCS, VCS_SET_VOL,
+			   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+		return;
+	}
+
+	tester_rsp(BTP_SERVICE_ID_VCS, VCS_SET_VOL, CONTROLLER_INDEX,
+		   BTP_STATUS_SUCCESS);
+}
+
+static void vol_up(void)
+{
+	int err;
+
+	LOG_DBG("Volume Up");
+
+	err = bt_vcp_vol_rend_vol_up();
+	if (err) {
+		tester_rsp(BTP_SERVICE_ID_VCS, VCS_VOL_UP,
+			   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+		return;
+	}
+
+	tester_rsp(BTP_SERVICE_ID_VCS, VCS_VOL_UP, CONTROLLER_INDEX,
+		   BTP_STATUS_SUCCESS);
+}
+
+static void vol_down(void)
+{
+	int err;
+
+	LOG_DBG("Volume Down");
+
+	err = bt_vcp_vol_rend_vol_down();
+	if (err) {
+		tester_rsp(BTP_SERVICE_ID_VCS, VCS_VOL_DOWN,
+			   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+		return;
+	}
+
+	tester_rsp(BTP_SERVICE_ID_VCS, VCS_VOL_DOWN, CONTROLLER_INDEX,
+		   BTP_STATUS_SUCCESS);
+}
+
+static void mute(void)
+{
+	int err;
+
+	err = bt_vcp_vol_rend_mute();
+	if (err) {
```

Comment:
Minor: Since the `err` value isn't logged or used in any way, we could do 

```suggestion
	if (bt_vcp_vol_rend_mute() != 0) {
```

The (perhaps better) alternative is to log the error so that it is easier to debug

Reply:
I've changed it so now we don't use `int err` as far as debugging is concerned when every function has a log in it - finding what fails shouldn't take much time.